drivers/vcluster/vcluster.go

```diff
@@ -656,3 +686,130 @@ func (v *VCluster) IsDeploymentHealthy(appNS string, deploymentName string, expe
+	secretValue := "vcluster-tests-secret"
+	// Create a new secret object
+	secret := &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: PxNamespace,
+		},
+		StringData: map[string]string{
+			ClusterWideSecretKey: secretValue,
+		},
+		Type: corev1.SecretTypeOpaque,
+	}
+	if _, err := k8sCore.CreateSecret(secret); err != nil {
+		return err
+	}
+	return nil
+}
+
+// DeleteSecret deletes a secret
+func DeleteSecret(name, namespace string) error {
+	return k8sCore.DeleteSecret(name, namespace)
+}
+
+// CreateVolumeSnapshotClass Creates a VolumeSnapshotclass in Vcluster context
+func (v *VCluster) CreateVolumeSnapshotClass(snapClassName string, params map[string]string) error {
+	vsc := &snapshotv1.VolumeSnapshotClass{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: snapClassName,
+		},
+		Driver:         k8s.CsiProvisioner,
+		DeletionPolicy: snapshotv1.VolumeSnapshotContentDelete,
+		Parameters:     params,
+	}
+	if _, err := v.SnapClient.SnapshotV1().VolumeSnapshotClasses().Create(context.TODO(), vsc, metav1.CreateOptions{}); err != nil {
+		return err
+	}
+	log.Infof("VolumeSnapshotClass %s created successfully", snapClassName)
+	return nil
+}
+
+// CreateSnapshot creates a volume snapshot for a given PVC in vcluster context
+func (v *VCluster) CreateSnapshot(snapshotName, pvcName, snapClassName, namespace string) error {
+	vs := &snapshotv1.VolumeSnapshot{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      snapshotName,
+			Namespace: namespace,
+		},
+		Spec: snapshotv1.VolumeSnapshotSpec{
+			VolumeSnapshotClassName: &snapClassName,
+			Source: snapshotv1.VolumeSnapshotSource{
+				PersistentVolumeClaimName: &pvcName,
+			},
+		},
+	}
+	_, err := v.SnapClient.SnapshotV1().VolumeSnapshots(namespace).Create(context.TODO(), vs, metav1.CreateOptions{})
+	if err != nil {
+		return err
+	}
+	log.Infof("Snapshot %s created successfully", snapshotName)
+	return nil
+}
+
+// RestorePVCFromSnapshot restores a CSI PVC from a volume snapshot
+func (v *VCluster) RestorePVCFromSnapshot(restoredPvcName, snapshotName, appNS, storageClassName, accessMode string) error {
+	var restoredAccessMode corev1.PersistentVolumeAccessMode
+	switch accessMode {
+	case "RWO":
+		restoredAccessMode = corev1.ReadWriteOnce
+	case "RWX":
+		restoredAccessMode = corev1.ReadWriteMany
+	case "ROX":
+		restoredAccessMode = corev1.ReadOnlyMany
+	default:
+		restoredAccessMode = corev1.ReadWriteOnce
+	}
+	restorePVC := &corev1.PersistentVolumeClaim{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      restoredPvcName,
+			Namespace: appNS,
+			Annotations: map[string]string{
+				"snapshot.alpha.kubernetes.io/snapshot": snapshotName,
+			},
+		},
+		Spec: corev1.PersistentVolumeClaimSpec{
+			StorageClassName: &storageClassName,
+			AccessModes:      []corev1.PersistentVolumeAccessMode{restoredAccessMode},
+			Resources: corev1.ResourceRequirements{
+				Requests: corev1.ResourceList{
+					corev1.ResourceStorage: resource.MustParse("5Gi"),
```

Comment:
Why `5Gi` is hardcoded here?

Reply:
Yes, this will become parameterized in the next PR when we enable an official Jenkins job